cookiecutter/prompt.py

```diff
@@ -198,41 +201,69 @@ def prompt_for_config(context, no_input=False):
-                val = render_variable(env, raw, cookiecutter_dict)
-
-                if not no_input:
-                    val = read_user_variable(key, val)
-
-                cookiecutter_dict[key] = val
+            cookiecutter_dict = get_cookiecutter_values(
+                key, raw, cookiecutter_dict, meta_information
+            )
         except UndefinedError as err:
             msg = "Unable to render variable '{}'".format(key)
             raise UndefinedVariableInTemplate(msg, err, context)
 
     # Second pass; handle the dictionaries.
     for key, raw in context['cookiecutter'].items():
+        # Check for conditional keywords.
+        if key.startswith(conditional_keyword):
+            conditional_value_to_check = key[conditional_keyword_length:]
+            if cookiecutter_dict[conditional_value_to_check] == 'yes':
+                for sub_key, sub_value in raw.items():
+                    cookiecutter_dict = get_cookiecutter_values(
+                        sub_key, sub_value, cookiecutter_dict, meta_information
+                    )
+                cookiecutter_dict.pop(key)
+            else:
+                cookiecutter_dict.pop(key)
         # Skip private type dicts not ot be rendered.
-        if key.startswith('_') and not key.startswith('__'):
+        elif key.startswith('_') and not key.startswith('__'):
             continue
 
-        try:
-            if isinstance(raw, dict):
-                # We are dealing with a dict variable
-                val = render_variable(env, raw, cookiecutter_dict)
+        else:
+            try:
+                if isinstance(raw, dict):
+                    # We are dealing with a dict variable
+                    val = render_variable(env, raw, cookiecutter_dict)
 
-                if not no_input and not key.startswith('__'):
-                    val = read_user_dict(key, val)
+                    if not no_input and not key.startswith('__'):
+                        val = read_user_dict(key, val)
 
-                cookiecutter_dict[key] = val
-        except UndefinedError as err:
-            msg = "Unable to render variable '{}'".format(key)
-            raise UndefinedVariableInTemplate(msg, err, context)
+                    cookiecutter_dict[key] = val
+            except UndefinedError as err:
+                msg = "Unable to render variable '{}'".format(key)
+                raise UndefinedVariableInTemplate(msg, err, context)
+
+    return cookiecutter_dict
+
+
+def get_cookiecutter_values(key, value, cookiecutter_dict, meta_information):
+    """Get cookiecutter values from keys.
+
+    :param key: key to retrieve value for
+    :param value: raw value to get
+    :param cookiecutter_dict: current cookiecutter dict
+    :param meta_information: information about this session
+
+    """
+    env = meta_information.get("env")
+    no_input = meta_information.get("no_input")
+    if isinstance(value, list):
```

Comment:
```suggestion
    if isinstance(value, (tuple, list)):
```

just to allow tuples too